dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -0,0 +1,220 @@
+		}
+		${!isCopied &&
+		`:hover {
+			color: ${themePalette('--share-button-liveblog-mobile')};
+			background-color: ${palette.neutral[100]};
+			svg {
+				fill: ${themePalette('--share-button-liveblog-mobile')};
+			}
+		}`}
+	}
+`;
+
+const getUrl = ({
+	pageId,
+	CMP,
+	blockId,
+}: {
+	pageId: string;
+	CMP?: string;
+	blockId?: string;
+}) => {
+	const searchParams = new URLSearchParams({});
+	if (CMP) searchParams.append('CMP', CMP); // Do we want to track this?
+	if (blockId) searchParams.append('page', `with:block-${blockId}`);
+
+	const blockHash = blockId ? `#block-${blockId}` : '';
+	const paramsString = searchParams.toString();
+	return new URL(
+		`${pageId}${paramsString ? '?' + paramsString : ''}${blockHash}`,
+		'https://www.theguardian.com/',
+	).href;
+};
+
+export const NativeShareButton = ({
+	onShare,
+	size,
+	isLiveBlogMeta,
+}: {
+	onShare: () => Promise<void>;
+	size?: Size;
+	isLiveBlogMeta: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+
+	return (
+		<Button
+			onClick={() => onShare().catch(() => {})}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={<SvgShare />}
+			css={[
+				buttonStyles,
+				nativeShare(sizeXSmall),
+				isLiveBlogMeta && liveBlogMobile(false),
+			]}
+		>
+			Share
+		</Button>
+	);
+};
+
+export const CopyLinkButton = ({
+	onShare,
+	size,
+	isLiveBlogMeta,
+	isCopied,
+}: {
+	onShare: () => Promise<void>;
+	size?: Size;
+	isLiveBlogMeta: boolean;
+	isCopied: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+	return (
+		<Button
+			onClick={() => onShare().catch(() => {})}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={isCopied ? <SvgCheckmark /> : <LinkIcon />}
+			css={[
+				...(isCopied
+					? [copiedButtonStyles(sizeXSmall)]
+					: [buttonStyles]),
+				sharedButtonStyles(sizeXSmall),
+				isLiveBlogMeta && liveBlogMobile(isCopied),
+			]}
+		>
+			{isCopied ? 'Link copied' : 'Copy link'}
+		</Button>
+	);
+};
+
+export const ShareButton = ({
+	size = 'small',
+	pageId,
+	blockId,
+	webTitle,
+	format,
+	context,
+}: Props) => {
+	const [isCopied, setIsCopied] = useState(false);
+	const [isShareSupported, setIsShareSupported] = useState(false);
+
+	const isLiveBlogMeta =
+		(format.design === ArticleDesign.LiveBlog &&
+			context === 'ArticleMeta') ||
+		context === 'SubMeta';
+
+	useEffect(() => {
+		setIsShareSupported(
+			typeof navigator !== 'undefined' && 'share' in navigator,
+		);
+	}, []);
+
+	const onShare = async () => {
+		if (isShareSupported) {
+			await navigator.share({
+				title: `${webTitle}`,
+				text: `${webTitle}:`,
+				url: getUrl({
+					pageId,
+					blockId,
+				}),
+			});
+		} else if ('clipboard' in navigator) {
+			await navigator.clipboard.writeText(
+				getUrl({
+					pageId,
+					blockId,
+				}),
+			);
+			setIsCopied(true);
+			setTimeout(() => setIsCopied(false), 3000);
+		}
+	};
```

Comment:
Rather than having a function that handles both of these behaviours,
sould we instead inline them as part of the `onClick` handler of each of the buttons themselves?

```tsx
eturn isShareSupported ? (
		<NativeShareButton
			onShare={() => {
			  void navigator.share({
				title: `${webTitle}`,
				text: `${webTitle}:`,
				url: getUrl({
					pageId,
					blockId,
				}),
			});
			}}
			size={size}
			isLiveBlogMeta={isLiveBlogMeta}
		/>
	) : (
		<CopyLinkButton
			onShare={() => {
			  if (!('clipboard' in navigator)) return;
			  navigator.clipboard.writeText(
				getUrl({
					pageId,
					blockId,
				}).then(() => {
				  setIsCopied(true);
                }),
			);
			}}
			size={size}
			isCopied={isCopied}
			isLiveBlogMeta={isLiveBlogMeta}
		/>
	);
```

As for a resetting copied value, I think it’s best to do this in a `useEffect`:

```ts
useEffect(() => {
  if(!isCopied) return; // nothing to do if already false
  const timer = setTimeout(() => {
    setIsCopied(false)
  }, 3000);
  // ensure that reruns of this effect clear the timeout
  return () => clearTimeout(timer);
}, [isCopied]) // ensure the dependencies are added to the array
```